index.js

```diff
@@ -0,0 +1,304 @@
+          core.addPath(newPath);
+        }
+      } else {
+        core.exportVariable(key, value);
+      }
+      core.info(`Set ${key}`);
+    }
+
+    // Create bin directory for wrapper scripts
+    const binDir = path.join(toolsPath, "bin");
+    await fs.promises.mkdir(binDir, { recursive: true });
+
+    // Create wrapper scripts based on parsed commands
+    for (const [cmd, fullPath] of Object.entries(commands)) {
+      const wrapperPath = path.join(binDir, cmd);
+      if (process.platform === "win32") {
+        // Create .cmd file for Windows
+        const cmdContent = `@echo off\r\n${fullPath} %*`;
+        await fs.promises.writeFile(wrapperPath + ".cmd", cmdContent);
+      } else {
+        // Create shell script for Unix
+        const shContent = `#!/bin/bash\n${fullPath} "$@"`;
+        await fs.promises.writeFile(wrapperPath, shContent);
+        await exec.exec("chmod", ["+x", wrapperPath]);
+      }
+    }
+
+    // Add bin directory to PATH
+    core.addPath(binDir);
+    core.info(
+      "ESP-IDF installation and environment setup completed successfully"
+    );
+  } catch (error) {
+    core.setFailed(error.message || "An unexpected error occurred");
+  }
+}
+
+// Function to fetch latest release version from GitHub API
+async function getLatestEimVersion() {
+  return new Promise((resolve, reject) => {
+    const options = {
+      hostname: "api.github.com",
+      path: "/repos/espressif/idf-im-cli/releases/latest",
+      headers: {
+        "User-Agent": "GitHub-Action-ESP-IDF-Setup",
+      },
+    };
+
+    const req = https.get(options, (res) => {
+      let data = "";
+
+      res.on("data", (chunk) => {
+        data += chunk;
+      });
+
+      res.on("end", () => {
+        if (res.statusCode === 200) {
+          try {
+            const release = JSON.parse(data);
+            resolve(release.tag_name);
+          } catch (error) {
+            reject(new Error("Failed to parse GitHub API response"));
+          }
+        } else {
+          reject(
+            new Error(`GitHub API request failed with status ${res.statusCode}`)
+          );
+        }
+      });
+    });
+
+    req.on("error", (error) => {
+      reject(error);
+    });
+
+    req.end();
+  });
+}
+
+function getEimDownloadUrl(platform, arch, version) {
+  const baseUrl = "https://github.com/espressif/idf-im-cli/releases/download";
+
+  switch (platform) {
+    case "linux":
+      return `${baseUrl}/${version}/eim-${version}-linux-${
+        arch === "arm64" ? "arm64" : "x64"
+      }.zip`;
+    case "darwin":
+      return `${baseUrl}/${version}/eim-${version}-macos-${
+        arch === "arm64" ? "aarch64" : "x64"
+      }.zip`;
+    case "win32":
+      return `${baseUrl}/${version}/eim-${version}-windows-x64.zip`;
+    default:
+      throw new Error(`Unsupported platform: ${platform}`);
+  }
+}
+
+async function installDependencies(platform) {
+  switch (platform) {
+    case "linux":
```

Comment:
What will happen if user runs it on a custom Linux runner without APT? We do not have to support that scenario, but maybe a nice error explaining what we do support?

Reply:
Good point. I was focused on the official GitHub runners but we should have a fallback.

Reply:
```
"---------------   WARNING   ---------------\n" +
"This action currently supports only official GitHub-hosted Ubuntu runners. " +
"If you're using a self-hosted runner or a different Linux distribution, " +
"please ensure all required dependencies are pre-installed."
---